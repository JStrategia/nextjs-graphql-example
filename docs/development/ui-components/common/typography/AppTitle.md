# AppTitle

Renders a `h1` tag with theme style and given text content.

## Usage

```tsx
function MyComponent() {
  return <AppTitle>Some text</AppTitle>;
}
```
