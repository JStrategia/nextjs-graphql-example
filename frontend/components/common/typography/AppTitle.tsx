import styled, { css } from 'styled-components';
import Typography from './Typography';

const AppTitle = styled.h1(
  ({ theme }) => css`
    ${Typography};
    text-align: center;
    font-size: ${theme.fonts.sizes.xxl};
  `,
);

export default AppTitle;
