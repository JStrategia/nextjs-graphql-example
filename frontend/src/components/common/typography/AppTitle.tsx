import styled, { css } from 'styled-components';
import Typography from 'components/common/typography/Typography';
import { Theme } from 'styles/theme';

const AppTitle = styled.h1(
  ({ theme }: { readonly theme: Theme }) => css`
    ${Typography};
    text-align: center;
    font-size: ${theme.fonts.sizes.xxl};
  `,
);

export default AppTitle;
