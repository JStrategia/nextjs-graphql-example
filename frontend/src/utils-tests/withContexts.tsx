import React from 'react';
import ThemeProvider from 'contexts/ThemeProvider';

function withContexts(children: React.ReactNode) {
  return <ThemeProvider>{children}</ThemeProvider>;
}

export default withContexts;
