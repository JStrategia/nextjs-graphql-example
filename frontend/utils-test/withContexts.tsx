import React, { ReactNode } from 'react';
import ThemeProvider from '../contexts/ThemeProvider';

function withContexts(children: ReactNode) {
  return <ThemeProvider>{children}</ThemeProvider>;
}

export default withContexts;
