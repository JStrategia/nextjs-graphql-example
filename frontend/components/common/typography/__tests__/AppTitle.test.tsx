import * as React from 'react';
import { create } from 'react-test-renderer';
import AppTitle from '../AppTitle';
import withContexts from '../../../../utils-test/withContexts';

describe('AppTitle', () => {
  describe('Given no text', () => {
    it('Should match snapshot', () => {
      const render = create(withContexts(<AppTitle />)).toJSON();

      expect(render).toMatchSnapshot();
    });
  });

  describe('Given text', () => {
    const text = 'some text';

    it('Should match snapshot', () => {
      const render = create(withContexts(<AppTitle>{text}</AppTitle>)).toJSON();

      expect(render).toMatchSnapshot();
    });
  });
});
