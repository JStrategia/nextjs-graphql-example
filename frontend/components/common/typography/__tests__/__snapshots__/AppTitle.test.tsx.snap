// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppTitle Given no text Should render 1`] = `
<h1
  className="sc-bczRLJ eNbHFO"
/>
`;

exports[`AppTitle Given text Should render 1`] = `
<h1
  className="sc-bczRLJ eNbHFO"
>
  some text
</h1>
`;
